index.js

```diff
@@ -133,6 +138,55 @@ Shopify.prototype.request = function request(url, method, key, params) {
   });
 };
 
+/**
+ * Updates GraphQL API call limits.
+ *
+ * @param {String} throttle The status returned in the GraphQL response
+ * @private
+ */
+Shopify.prototype.updateGqlLimits = function updateGqlLimits(throttle) {
+  const limits = this.callGraphqlLimits;
+
+  limits.remaining = throttle.currentlyAvailable;
+  limits.current = throttle.maximumAvailable - throttle.currentlyAvailable;
+  limits.max = throttle.maximumAvailable;
+
+  this.emit('callGraphqlLimits', limits);
+};
+
+/**
+ * Sends a request to the Shopify GraphQL API endpoint.
+ *
+ * @param {Object} [data] Request body
+ * @return {Promise}
+ * @public
+ */
+Shopify.prototype.graphql = function graphql(data) {
+  const url = assign({ path: '/admin/api/graphql.json' }, this.baseUrl);
+  const options = assign({
+    headers: {
+      'User-Agent': `${pkg.name}/${pkg.version}`,
+      'Content-Type': 'application/graphql'
+    },
+    timeout: this.options.timeout,
+    retries: 0,
+    method: 'POST',
+    body: data
```

Comment:
Is building `data` user responsibility? Can the user use variables? If so is the following signature reasonable?

```
shopify.graphql(query[, variables])
```

I think the current interface is not very appealing. I mean what is the advantage of using this module instead of got or node-fetch directly?

Reply:
I know, but this a direct request by Shopify and they want it like that, at least by now. We can think of an improvement later on, but they are rushing with this.

Reply:
Ok can you update the TypeScript definitions to include this new method?

Reply:
Done